examples/dgmulti_2d/elixir_navierstokes_convergence_curved.jl

```diff
@@ -0,0 +1,212 @@
+                                        + 2.0   * rho  * v1 * v1_x
+                                        + rho_y * v1   * v2
+                                        + rho   * v1_y * v2
+                                        + rho   * v1   * v2_y
+    # stress tensor from x-direction
+                      - 4.0 / 3.0 * v1_xx * mu_
+                      + 2.0 / 3.0 * v2_xy * mu_
+                      - v1_yy             * mu_
+                      - v2_xy             * mu_ )
+  # y-momentum equation
+  du3 = ( rho_t * v2 + rho * v2_t + p_y + rho_x * v1    * v2
+                                        + rho   * v1_x  * v2
+                                        + rho   * v1    * v2_x
+                                        +         rho_y * v2^2
+                                        + 2.0   * rho   * v2 * v2_y
+    # stress tensor from y-direction
+                      - v1_xy             * mu_
+                      - v2_xx             * mu_
+                      - 4.0 / 3.0 * v2_yy * mu_
+                      + 2.0 / 3.0 * v1_xy * mu_ )
+  # total energy equation
+  du4 = ( E_t + v1_x * (E + p) + v1 * (E_x + p_x)
+              + v2_y * (E + p) + v2 * (E_y + p_y)
+    # stress tensor and temperature gradient terms from x-direction
+                                - 4.0 / 3.0 * v1_xx * v1   * mu_
+                                + 2.0 / 3.0 * v2_xy * v1   * mu_
+                                - 4.0 / 3.0 * v1_x  * v1_x * mu_
+                                + 2.0 / 3.0 * v2_y  * v1_x * mu_
+                                - v1_xy     * v2           * mu_
+                                - v2_xx     * v2           * mu_
+                                - v1_y      * v2_x         * mu_
+                                - v2_x      * v2_x         * mu_
+         - T_const * inv_rho_cubed * (        p_xx * rho   * rho
+                                      - 2.0 * p_x  * rho   * rho_x
+                                      + 2.0 * p    * rho_x * rho_x
+                                      -       p    * rho   * rho_xx ) * mu_
+    # stress tensor and temperature gradient terms from y-direction
+                                - v1_yy     * v1           * mu_
+                                - v2_xy     * v1           * mu_
+                                - v1_y      * v1_y         * mu_
+                                - v2_x      * v1_y         * mu_
+                                - 4.0 / 3.0 * v2_yy * v2   * mu_
+                                + 2.0 / 3.0 * v1_xy * v2   * mu_
+                                - 4.0 / 3.0 * v2_y  * v2_y * mu_
+                                + 2.0 / 3.0 * v1_x  * v2_y * mu_
+         - T_const * inv_rho_cubed * (        p_yy * rho   * rho
+                                      - 2.0 * p_y  * rho   * rho_y
+                                      + 2.0 * p    * rho_y * rho_y
+                                      -       p    * rho   * rho_yy ) * mu_ )
+
+  return SVector(du1, du2, du3, du4)
+end
```

Comment:
Is this referring to the type-unstable (indexing) comment above?

Reply:
No - to the actual formulas of the manufactured solution

Reply:
It's the exact same as the setup for `elixir_navierstokes_convergence.jl`. I can add a comment mentioning this?

Reply:
Alternatively, I could just run `elixir_navierstokes_convergence.jl` and pass a new mesh through `trixi_include` instead of creating new elixirs.

Reply:
I switched back to running `elixir_navierstokes_convergence.jl` and using `trixi_include` to pass a new mesh in adc0fdec01d75c45c82f3984d6cbf728e846cf60

Reply:
Nevermind, using `trixi_include` didn't work. I added some comments noting the origin of the manufactured solution.

Reply:
Alternatively... shouldn't we add the NSE convergence test to the `CompressibleNSEDiffusionXD` types? IIRC, we do have a similar function for most other equation systems. Or was there an issue with that why this didn't work/why this is super awkward?

Reply:
It's fine from my side to keep it as it is - I just wanted to say that I didn't want to read it :sweat_smile:

Reply:
@sloede we could. The only potential issue I see is that the source and initial condition require functions `mu()` and `prandtl_number()` to be defined (see https://github.com/jlchan/Trixi.jl/blob/a8c0933bd0077c45372c3937c40fce772e348886/examples/dgmulti_2d/elixir_navierstokes_convergence.jl#L56-L57). These functions are necessary due to the fact that initial conditions cannot specialize on only one equation type, since they are called both with `equations::CompressibleEulerEquations2D` and `equations::CompressibleNavierStokesDiffusion2D` in different places. 

To me, moving the NSE convergence test functions out of the elixir makes these implicit requirements even less clear.

Reply:
Good point. Let's keep them as they are, please.